prometheus/rootfs/bin/boot.sh

```diff
@@ -22,18 +22,23 @@ done
 
 set +e
 
-OPTS=""
+until [ -n "${DEIS_MONITOR_ALERT_SERVICE_HOST+1}" ]; do
```

Comment:
Yep, that is correct, let me 'splain my comment better. If the alert manager is up *before* prometheus starts and it is in the same namespace, then the alertmanagers service env variables will be set and can be access by prometheus.

Thing is, though, that the `until` loop wont work because if prometheus comes up before alertmanager, the env vars will never become available to prometheus - they arent injected or anything. The env thats set when it comes up is the env its gets, afaik. It would be awesome if k8s changed that.